### PR TITLE
perf(signal): reuse leases across ciphertext APIs

### LINKED ISSUE #1040: Use counter leases in every ciphertext-producing API
The main send path uses `persist_signal_state_pre_wire`, so a synchronous flush happens only when an outbound lease is raised. A few other APIs still flush the entire Signal cache after every encryption:

- [public group encryption](https://github.com/oxidezap/whatsapp-rust/blob/0e1db5762aff1988f74f95a6dd614bcba0c5d644/src/features/signal.rs#L205-L219)
- [participant-node creation](https://github.com/oxidezap/whatsapp-rust/blob/0e1db5762aff1988f74f95a6dd614bcba0c5d644/src/features/signal.rs#L320-L334)
- [VoIP call-key fanout](https://github.com/oxidezap/whatsapp-rust/blob/0e1db5762aff1988f74f95a6dd614bcba0c5d644/src/voip/facade.rs#L465-L477)

Group sender keys now use the same 64-iteration lease model as pairwise sessions, so these full flushes are no longer required for every call. They also serialize unrelated dirty Signal state and add storage contention to APIs that may sit directly on an interactive send/call path.

After releasing the session or sender-key locks, these paths should use `persist_signal_state_pre_wire().await?`. A lease boundary still blocks on durable storage; an advance already covered by a durable lease schedules the existing coalesced write-behind. Decrypt and explicit durability/control APIs should keep their current full-flush behavior.

This can remove up to 63 synchronous flushes per 64-iteration lease without weakening crash safety. It should reduce SQLite transactions, cache-wide serialization and tail latency during group sends and call setup. The stale comments that still say sender keys have no lease should be updated in the same change.

Please add flush-count tests for boundary versus in-lease calls and an A/B benchmark for warm group encryption and multi-device call-key fanout.

### PR DESCRIPTION
**Pull request by jlucaso1**
## Summary

- Route public group encryption, participant-node fanout, and VoIP call-key fanout through the pre-wire lease gate after releasing their Signal locks.
- Keep lease-boundary writes synchronous and fail closed, while lease-covered advances use the existing coalesced write-behind. Decrypt and explicit durability paths keep their full flushes.
- Add write-count regressions for sender-key boundaries, warm participant fanout, and warm VoIP fanout.

## Safety

A ciphertext at a new counter boundary is still returned only after its raised lease is durable. A covered ciphertext can recover after a crash by skipping to the already-persisted ceiling, so it cannot reuse a message key or IV. Running the gate after the per-session or sender-key locks also preserves the existing lock ordering.

## Measured impact

A/B against `c63d949d` with the same real-SQLite harness and alternating runs:

- Warm group encryption, 1,024 calls/sample (`n=16`): **84.69 ms → 23.98 ms** (**-71.7%**).
- Warm participant fanout, 4 recipients × 256 calls/sample (`n=12`): **49.16 ms → 8.69 ms** (**-82.3%**).
- DHAT group run, 4 × 1,024 calls: allocated bytes **19.63 MB → 10.32 MB** (-47.4%), allocation blocks **214,103 → 80,260** (-62.5%), process CPU **0.90 s → 0.47 s**. Peak live allocation was 0.99 MB → 1.12 MB and end-of-run live allocation 0.03 MB → 0.04 MB, so the win is reduced transaction/allocation churn rather than retained-memory reduction.

The boundary regression observes two sender-key batch writes across 65 encryptions: the initial reservation and the next 64-iteration boundary.

## Validation

- `cargo fmt --all -- --check`
- `cargo clippy --all-targets -- -D warnings`
- `cargo clippy --workspace --exclude e2e-tests --all-features --all-targets -- -D warnings`
- `cargo test --all` (including the complete mock-server E2E suite)
- `cargo test -p wacore store::signal_cache -- --nocapture`
- VoIP lease regression under `voip tokio-native tokio-transport`
- Both wasm32 release builds from `.github/workflows/wasm.yml`

Closes #1040

### CODE REVIEW

File: wacore/src/store/in_memory.rs [hunk 1]
```diff
@@ -342,6 +354,31 @@ impl SignalStore for InMemoryBackend {
         Ok(())
     }
 
+    async fn put_sender_keys_batch(&self, sender_keys: &[(Arc<str>, Bytes)]) -> Result<()> {
+        #[cfg(any(test, feature = "test-util"))]
+        {
+            self.sender_key_batch_writes.fetch_add(1, Ordering::Relaxed);
+            if self.fail_sender_key_writes.load(Ordering::Relaxed) {
```
[the remark this answers is not in the document]
Comment (jlucaso1) [reply wacore/src/store/in_memory.rs:363]: Clarified both batch-counter getters: they intentionally count attempts, including injected failures, so failure-path tests can prove the pre-wire durability gate ran. This keeps sender-key and session hooks consistent.
